qa/L0_lifecycle/lifecycle_test.py

```diff
@@ -2885,6 +2885,21 @@ def test_concurrent_load_unload(self):
         self.assertTrue(triton_client.is_server_live())
         self.assertTrue(triton_client.is_server_ready())
         self.assertFalse(triton_client.is_model_ready("identity_zero_1_int32"))
+        # Unload identity_zero_1_int32 and immediately load it
+        # The model can either be loaded or unloaded but server must not crash
```

Comment:
In the case you try to hit, doesn't the order matter? i.e. load immediately after an unload. How often does this test catch the failure without your earlier fix?

Reply:
Yes, the order matters. It is refactored into a separate case to have better checking.

If `load -> unload`: The load will block the unload, which the model is unloaded at the end.
If `unload -> load`: The unload will block the load, which the model is loaded at the end.

Since the unload is async, if it is performed at the end, a poll/timeout is needed to ensure the unload is completed such that `is_model_ready` produces the correct result.

Without the earlier fix, the chance of catching it is about 1/50.

Reply:
> Without the earlier fix, the chance of catching it is about 1/50.

This may not be a good test case then.. Or at least you need to put emphasize that one must look into the failure when this particular test fails. Otherwise, I can see when something actually breaks, and luckily get caught by this test, one may just re-run the test and say it is just turbulence unless the re-run hits another 1/50 chance.

Reply:
Rewrite the test to perform stress test on load and unload API instead. The rewritten test failed 10/10 without the earlier fix, and passed 10/10 with the earlier fix.